contracts/spell/AuraSpell.sol

```diff
@@ -277,7 +284,7 @@ contract AuraSpell is IAuraSpell, BasicSpell {
 
         for (i; i < length; ++i) {
             if (tokens[i] != lpToken) {
-                amountsIn[j] = IERC20(tokens[i]).balanceOf(address(this));
+                amountsIn[j] = tokens[i] == borrowToken ? borrowAmount : 0;
                 if (amountsIn[j] > 0) {
                     IERC20(tokens[i]).universalApprove(vault, amountsIn[j]);
                     maxAmountsIn[i] = amountsIn[j];
```

Comment:
The `amounts` are already initialized to zero above, so I would simplify this as follows:
```suggestion
                if (tokens[i] == borrowToken) {
                    amountsIn[j] = borrowAmount;
                    IERC20(tokens[i]).universalApprove(vault, borrowAmount);
                    maxAmountsIn[i] = borrowAmount;
```